core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -757,7 +746,9 @@ object GlobalPSBTRecord {
   }
 
   case class Unknown(key: ByteVector, value: ByteVector)
-      extends GlobalPSBTRecord
+      extends GlobalPSBTRecord {
+    require(PSBTGlobalKeyId.fromKey(key) == PSBTGlobalKeyId.UnknownKeyId)
```

Comment:
nit: error message please (you can even make the `fromKey` into a `private val` and put it in the error message)